pkg/agent/memberlist/cluster.go

```diff
@@ -523,8 +546,8 @@ func (c *Cluster) ShouldSelectIP(ip, externalIPPool string, filters ...func(stri
 
 // SelectNodeForIP returns the closest item (Node name) in the hash to the provided key (IP) and ExternalIPPool.
 func (c *Cluster) SelectNodeForIP(ip, externalIPPool string, filters ...func(string) bool) (string, error) {
-	if externalIPPool == "" || ip == "" {
-		return "", fmt.Errorf("IP and externalIPPool cannot be empty")
+	if ip == "" {
```

Comment:
`ip` could be empty when the egress is not scheduable (non IP is allocated to it), we still need to select one Node/agent to update its status/